### ISSUE TITLE
Error on plan on aws when using template for user_data on launch conf.

### ISSUE THREAD

**Issue by aldarund** [hashicorp/terraform#4551]
I have following launch conf.

```
resource "aws_launch_configuration" "ecs" {
  name_prefix          = "ecs-rentapp-launch-"
  image_id             = "${lookup(var.ecs_amis, var.AWS_REGION)}"
  /* @todo - split out to a variable */
  instance_type        = "${var.ecs_instance_type}"
  key_name             = "${aws_key_pair.deployer.key_name}"
  iam_instance_profile = "${aws_iam_instance_profile.ecs.id}"
  security_groups      = ["${aws_security_group.ecs_node.id}"]
  user_data            = "${template_file.launch_user_data.rendered}"
  lifecycle {
      create_before_destroy = true
  }
}
resource "template_file" "launch_user_data" {
  template = "${file("user-data/ecs-launch-user-data.sh")}"
}
```

I changed file user-data/ecs-launch-user-data.sh and ecs_amis var. After this plan not working anymore. Following error occuring:

Error running plan: 1 error(s) occurred:
- Cycle: template_file.launch_user_data, aws_launch_configuration.ecs, aws_autoscaling_group.ecs, aws_launch_configuration.ecs (destroy), template_file.launch_user_data (destroy)

When i changed user_data to inline string instead of template , plan ( and apply) completed with success.

**Comment by jen20** [comment]
Hi @aldarund! If you add:

```
lifecycle {
      create_before_destroy = true
}
```
to the template, does that resolve your issue?

**Comment by aldarund** [comment]
yes

**Comment by vitalis** [comment]
I have the same issue, if I add 
```
lifecycle {
      create_before_destroy = true
}
```
it forces new resource on both `template_file` and `aws_launch_configuration`
```
-/+ aws_launch_configuration.ecs_staging
    associate_public_ip_address:               "true" => "1"
    ebs_block_device.#:                        "0" => "<computed>"
    ebs_optimized:                             "false" => "<computed>"
    enable_monitoring:                         "true" => "1"
    iam_instance_profile:                      "ecs-instance-profile" => "ecs-instance-profile"
    image_id:                                  "ami-cbfdb2a1" => "ami-cbfdb2a1"
    instance_type:                             "c4.xlarge" => "c4.xlarge"
    key_name:                                  "applicaster-052015" => "applicaster-052015"
    name:                                      "terraform-n7tp25vvafg6zci4sbphs5fjaq" => "<computed>"
    root_block_device.#:                       "1" => "1"
    root_block_device.0.delete_on_termination: "true" => "1"
    root_block_device.0.iops:                  "0" => "<computed>"
    root_block_device.0.volume_size:           "100" => "100"
    root_block_device.0.volume_type:           "gp2" => "gp2"
    security_groups.#:                         "2" => "2"
    security_groups.230592924:                 "sg-b56d31d1" => "sg-b56d31d1"
    security_groups.3742176214:                "sg-ab6d31cf" => "sg-ab6d31cf"
    user_data:                                 "7efab11f81fa6752a866c2264720889e4be7b60a" => "08118b35c10c706c61c9c9d7888946823b46198b" (forces new resource)
```

**Comment by mitchellh** [comment]
This should be fixed in TF 0.8!
